Library/Formula/fbthrift-compiler.rb

```diff
@@ -0,0 +1,92 @@
+class FbthriftCompiler < Formula
+  desc "IDL compiler from Facebook's Thrift, an RPC system"
+  homepage "https://github.com/facebook/fbthrift"
+  url "https://github.com/facebook/fbthrift/archive/v0.28.0.tar.gz"
+  version "28.0"
+  sha256 "1aa577d313c24950be1f0bb9681c4c959ce010a7ba298870c3f9a5b54c6cdc61"
+
+  head "https://github.com/facebook/fbthrift.git"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+
+  depends_on "glog"
+  depends_on "boost"
+  depends_on "folly"
+
+  # 1. Thrift's configure.ac conflates the cpp runtime with the compiler. Even
+  # if we specify `--without-cpp` to skip the checks for e.g. numa which isn't
+  # available on OSX, the configure ends up forcing WITH_CPP=true anyway which
+  # tries to build the cpp runtime in lib.  We work around that by explicitly
+  # removing lib from the list of SUBDIRS to process.
+  # 2. The build process for the python chunk of the compiler doesn't really
+  # work too well. But that's ok, because it's optional, and the cpp half
+  # doesn't know how to call the python half on OSX anyway. So exclude that
+  # too.
+  # For details, see:
+  #  https://github.com/facebook/fbthrift/issues/102
+  #  https://github.com/facebook/fbthrift/issues/62
+  #  https://github.com/facebook/fbthrift/issues/50
+  #  https://github.com/facebook/fbthrift/issues/21
+  patch :p1, :DATA
```

Comment:
Any indication at all when these will be fixed upstream? We're usually fairly hesitant to carry new formulae that require patching if those patches haven't been accepted.